lib/elixir/lib/option_parser.ex

```diff
@@ -118,6 +125,36 @@ defmodule OptionParser do
   end
 
   @doc """
+  The same as `parse/2` but raises an `OptionParser.InvalidOptionError`
+  exception if any invalid options are given.
+
+  If there weren't any errors, returns a three-element tuple as follows:
+
+      1. parsed options,
+      2. remaining arguments,
+      3. empty list.
+
+  ## Examples
+
+      iex> OptionParser.parse!(["--limit", "xyz"], strict: [limit: :integer])
+      ** (OptionParser.InvalidOptionError) 1 error found! Option --limit is of the wrong type, expected a integer, given "xyz"
+
+      iex> OptionParser.parse!(["--unknown", "xyz"], strict: [])
+      ** (OptionParser.InvalidOptionError) 1 error found! Unknown option --unknown
+
+      iex> OptionParser.parse!(["-l", "xyz", "-f", "bar"], strict: [limit: :integer, foo: :integer], aliases: [l: :limit, f: :foo])
+      ** (OptionParser.InvalidOptionError) 2 errors found! Option -l is of the wrong type, expected a integer, given "xyz". Option -f is of the wrong type, expected a integer, given "bar"
+
+  """
+  @spec parse!(argv, options) :: {parsed, argv, errors} | no_return
+  def parse!(argv, opts \\ []) when is_list(argv) and is_list(opts) do
+    case parse(argv, opts) do
+      {_parsed, _argv, []} = result -> result
```

Comment:
Maybe this function should return `{parsed, argv}` without returning any `errors` since I guess they'd always be `[]`, right?